zaza/openstack/charm_tests/nova/tests.py

```diff
@@ -178,6 +178,52 @@ def test_virsh_audit_action(self):
                     "The action failed: {}".format(action.data["message"]))
 
 
+class NovaCloudControllerActionTest(test_utils.OpenStackBaseTest):
+    """Run nova-cloud-controller specific tests.
+
+    Add this test class for new nova-cloud-controller action
+    to avoid breaking older version.
+    """
+
+    def test_sync_compute_az_action(self):
+        """Test sync-compute-availability-zones action."""
+        juju_units_az_map = {}
+        for unit in zaza.model.get_units('nova-compute',
+                                         model_name=self.model_name):
+            result = zaza.model.run_on_unit(unit.name,
+                                            'echo $JUJU_AVAILABILITY_ZONE',
+                                            model_name=self.model_name,
+                                            timeout=60)
+            self.assertEqual(int(result['Code']), 0)
+            juju_units_az_map[unit.public_address] = result['Stdout'].strip()
+            continue
+
+        session = openstack_utils.get_overcloud_keystone_session()
+        nova = openstack_utils.get_nova_session_client(session)
+        expected_output = ''
+        for hypervisor in nova.hypervisors.list():
+            az = juju_units_az_map.get(hypervisor.host_ip, None)
+            expected_output += ('Hypervisor {} added to '
+                                'availability zone {}\n'.format(
+                                    hypervisor.hypervisor_hostname, az))
+
```

Comment:
should we assert before running the action that the AZs are different? So that we're sure that the action is needed and doing something?

Reply:
I think checking the output itself is somewhat brittle. Discussing this with David, we reached the conclusion that we should just check that the action is successful since the subsequent code tests to make sure the aggregates are setup correctly.

Reply:
I agree with @wolsen on this one.